src/redux/sagas/utils/extendedInfuraProvider.ts

```diff
@@ -20,7 +20,7 @@ export class ExtendedInfuraProvider extends InfuraProvider {
       async () => {
         const result = await this._parentGetCode(addressOrName, blockTag);
         if (result === '0x') {
-          return undefined;
+          return '';
```

Comment:
Why this change from `undefined` to an empty `string` ?

Reply:
I saw that typescript was complaining about it returning `undefined` when according to the type definition of `getCode` it should return `Promise<String>`. 

So at first, I changed it to `Promise<String | undefined>` but then typescript would complain that the `Property 'getCode' in type 'ExtendedInfuraProvider' is not assignable to the same property in base type 'InfuraProvider'.` so, in the end, I left it as an empty string as that's still a falsy value.

This is also the case in the `ExtendedJsonRpcProvider`.

Do you think that this change could be a problem? It seems like this is an "internal" function as I didn't find places where it was used in the dapp

Reply:
Nah, don't worry. Leave it in for now.

Hopefully we'll be able to remove these "custom" functions entirely at some point